allennlp/models/archival.py

```diff
@@ -156,6 +157,46 @@ def archive_model(
                         archive.add(path, arcname=arcname)
 
 
+def load_from_hub(
+    identifier: str,
+    cuda_device: int = -1,
+    overrides: Union[str, Dict[str, Any]] = "",
+) -> Archive:
+    if "@" in identifier:
+        repo_id = identifier.split("@")[0]
+        revision = identifier.split("@")[1]
+    else:
+        repo_id = identifier
+        revision = None
+
+    serialization_dir = Path(CACHE_DIRECTORY) / repo_id
+    weights_path = serialization_dir / _WEIGHTS_NAME
+    config_path = serialization_dir / CONFIG_NAME
+
+    logger.warning(f"Pulling repository {repo_id}" + (f" on revision {revision}." if revision is not None else "."))
+    if os.path.exists(serialization_dir):
+        repo = Repository(serialization_dir)
+        repo.git_pull(rebase=True)
+    else:
+        repo = Repository(serialization_dir, f"https://huggingface.co/{repo_id}")
+
+    if revision is not None:
+        repo.git_fetch()
+        repo.git_reset(revision, hard=True)
```

Comment:
I think you'd want to do a `git checkout ${revision}` instead of a git reset --hard, no?

Reply:
Yes that would work too